ipa-core/src/protocol/dp/mod.rs

```diff
@@ -98,9 +128,59 @@ where
     )
     .await
     .unwrap();
+
     // Step 5 Transpose output representation
     Ok(Vec::transposed_from(&histogram_noised)?)
 }
+
+// dp_for_aggregation is currently where the DP parameters epsilon, delta
+// are introduced and then from those the parameters of the noise distribution to generate are
+// calculated for use in aggregating histograms.  The DP parameters query_epsilon and
+// per_user_credit_cap come as inputs to the query with per_user_sensitivity_cap = 2^{SS_BITS}
+/// # Errors
+/// will propogate errors from `apply_dp_noise`
+/// # Panics
+/// may panic from asserts down in  `gen_binomial_noise`
+/// may panic if running with DP noise but epsilon is not in the range (0,10].
```

Comment:
this does not match what is actually going on. Also I may prefer returning an error rather than panic here - you are reacting on the input you've been given and panic is not the best way to handle that imo

Reply:
okay, switched it to return an error for epsilon out of bounds.